### PR TITLE
Fix some issues on PHP 8.2

### PR DESCRIPTION
**Pull request by cedric-anne**
I found some issues while trying to run my test suite on PHP 8.2.

1. `['parent', 'mymethod']` is now deprecated, so I replaced it with `[parent::class, 'mymethod']`.
2. Usage of dynamic properties is now deprecated, so I added `atoum\atoum\report\fields\runner\event\cli\dot::$progressBar` declaration.
3. Casting an exception to array seems to be impossible now, so I changed the `atoum\atoum\tests\units\asserters\castToArray::testSetWith()` test to a variable that can be cast to array.

I added PHP 8.2 as experimental in the test suite.

### CODE REVIEW

File: tests/units/classes/asserters/castToArray.php [hunk 1]
```diff
@@ -47,10 +47,10 @@ public function testSetWith()
                     ->setAdapter($adapter)
             )
             ->then
-                ->object($asserter->setWith($object = new \exception()))->isIdenticalTo($asserter)
+                ->object($asserter->setWith($object = new \arrayiterator()))->isIdenticalTo($asserter)
```
Comment (Grummfy) [inline tests/units/classes/asserters/castToArray.php:50]: are you sure about this?

Comment (cedric-anne) [reply tests/units/classes/asserters/castToArray.php:50]: Casting the exception to array was trigerring an error in the PHP 8.2 build I used, so I had to found something that can be casted to array, in order to fix this test.

**Comment by cedric-anne** [comment]
@Grummfy PHP 8.2 RC1 should be released today. I do not want to hurry you, but I think having this PR merged on master branch may be usefull, so people that have to run their test suite on PHP 8.2 would be able to have these issues fixed and may probably find other issues, if there are any remaining.

**Comment by Grummfy** [comment]
sorry I completely forget the PR

**Pull request merged**